lib/io/hardware_gpio.cpp

```diff
@@ -2,13 +2,13 @@
 
 namespace hyped::io {
 
-std::optional<core::DigitalSignal> HardwareGpioReader::read()
+std::optional<core::DigitalSignal> HardwareGpioReader::readPin()
 {
   // TODOLater: implement
   throw -1;
 }
 
-core::Result HardwareGpioWriter::write(const core::DigitalSignal state)
+core::Result HardwareGpioWriter::writeToPin(const core::DigitalSignal state)
```

Comment:
not sure i agree with this rename? this function doesn't take a pin as an argument so why is it called `writeToPin`?

Reply:
`writeState` and `readState`?

Reply:
or signal instead of state

Reply:
but like... why not just `write`? I get that we want to make things descriptive but `brakes_pin_.write(core::DigitalSignal::kHigh)` doesn't look particularly hard to parse, for example?

Reply:
because the compiler confuses the linux `write` with the gpio `write` and thinks its a recursive function. not sure how to get around it with c libraries not having a namepsace -> was what I was thinking, just found out that global namespace is a thing